packages/graph/hash_graph/lib/graph/src/knowledge/mod.rs

```diff
@@ -13,30 +13,59 @@ pub use self::{
 };
 use crate::{
     ontology::{
-        PersistedDataType, PersistedEntityType, PersistedLinkType, PersistedPropertyType,
-        QueryDepth,
+        OntologyQueryDepth, PersistedDataType, PersistedEntityType, PersistedLinkType,
+        PersistedPropertyType,
     },
     store::query::Expression,
 };
 
-/// Query to read [`Entities`], which satisfy the [`Expression`].
+/// Distance to explore when querying a rooted subgraph on entities and links.
+///
+/// Entities may link to other entities through links. The depths provided alongside a query specify
+/// how many steps to explore along a chain of references _of a certain kind of type_. Meaning, any
+/// chain of entities and links will be resolved up to the given depth. These can be composed with
+/// [`OntologyQueryDepth`] to explore ontology types.
+///
+/// A depth of `0` means that links are explored for that specific kind of type.
+///
+/// **Note**: The concept is the same as the [`OntologyQueryDepth`] but it feels a little different
+/// as entities and links are chained in an alternate way, between every entity there is a
+/// link and vice versa.
+///
+/// # Example
+///
+/// - `Entity1` links to `Entity2` by `Link1`
+/// - `Entity2` links to `Entity3` by `Link2`
+///
+/// If a query on `Entity1` is made with the following depths:
+/// - `link_query_depth: 2`
+/// - `link_target_entity_query_depth: 1`
+///
+/// the query will resolve up to two links, but only a single entity:
+/// - `linkedEntities`: \[`Entity2`]
+/// - `links`: \[`Link1`, `Link2`]
+pub type KnowledgeQueryDepth = u8;
```

Comment:
Maybe we should call it KnowledgeGraphQueryDepth?

I don't feel strongly about this one

Reply:
It's good I guess, then it's consistent with `KnowledgeGraphQuery`: 695278dcca9caccdd59b309c856baaa1feda2d85